lib/ddtrace/contrib/active_record/patcher.rb

```diff
@@ -95,6 +100,25 @@ def self.sql(_name, start, finish, _id, payload)
         rescue StandardError => e
           Datadog::Tracer.log.error(e.message)
         end
+
+        def self.instantiation(_name, start, finish, _id, payload)
+          span_type = Datadog::Ext::SQL::TYPE
+
+          span = get_option(:tracer).trace(
+            'active_record.instantiation',
+            resource: payload.fetch(:class_name),
+            service: database_service,
```

Comment:
I think it should be the main service. If we set the `database_service`, it will be considered as a latency of your database service (let's say a MySQL Cluster), while what slows down is not the query itself, but ActiveRecord allocating objects.

Makes sense?

Reply:
Agreed. I thought the same thing (that there wasn't a neat category for this sort of thing. I'll make that change.

Reply:
Actually I don't think this is possible, since this ActiveRecord implementation doesn't know what a "main service" is. It might be Sinatra, but it could be a generic Ruby application, too. There's no good way to tell, short of adding an option for specifically configuring service name just for `active_record.instantiation`, e.g. something like `c.use :active_record, ruby_service_name: 'active_record'`.